### PR TITLE
Update docs/stargz.md: fix a typo for docker version

### PR DESCRIPTION
**Pull request by yankay**
Fix the https://github.com/containerd/nerdctl/issues/2320

### CODE REVIEW

File: docs/stargz.md [hunk 1]
```diff
@@ -176,8 +176,8 @@ You can use zstd compression with lazy pulling support (a.k.a zstd:chunked) inst
 - Cons
   - Old tools might not support. And unsupported by some tools yet.
     - zstd support by OCI Image Specification is still under rc (2022/11). will be added to [v1.1.0](https://github.com/opencontainers/image-spec/commit/1a29e8675a64a5cdd2d93b6fa879a82d9a4d926a).
-    - zstd support unreleased [by Docker](https://github.com/moby/moby/pull/41759/commits/e187eb2bb5f0c3f899fe643e95d1af8c57e89a73) (will be added to v22.06).
-    - [containerd >= v1.5](https://github.com/containerd/containerd/releases/tag/v1.5.0) supports zstd.
+    - zstd support by [docker >=v23.0.0](https://github.com/moby/moby/releases/tag/v23.0.0).
+    - zstd support by [containerd >= v1.5](https://github.com/containerd/containerd/releases/tag/v1.5.0).
```
Comment (fahedouch) [inline docs/stargz.md:180]:
```suggestion
     - zstd supported by [docker >=v23.0.0](https://github.com/moby/moby/releases/tag/v23.0.0).
     - zstd supported by [containerd >= v1.5](https://github.com/containerd/containerd/releases/tag/v1.5.0).
```

File: docs/stargz.md [hunk 2]
```diff
@@ -176,8 +176,8 @@ You can use zstd compression with lazy pulling support (a.k.a zstd:chunked) inst
 - Cons
   - Old tools might not support. And unsupported by some tools yet.
     - zstd support by OCI Image Specification is still under rc (2022/11). will be added to [v1.1.0](https://github.com/opencontainers/image-spec/commit/1a29e8675a64a5cdd2d93b6fa879a82d9a4d926a).
-    - zstd support unreleased [by Docker](https://github.com/moby/moby/pull/41759/commits/e187eb2bb5f0c3f899fe643e95d1af8c57e89a73) (will be added to v22.06).
-    - [containerd >= v1.5](https://github.com/containerd/containerd/releases/tag/v1.5.0) supports zstd.
+    - zstd support by [docker >=v23.0.0](https://github.com/moby/moby/releases/tag/v23.0.0).
```
Comment (fahedouch) [inline docs/stargz.md:179]:
```suggestion
    - zstd supported by [docker >=v23.0.0](https://github.com/moby/moby/releases/tag/v23.0.0).
```

Comment (yankay) [reply docs/stargz.md]: Thanks @fahedouch for the PR review

It has been changed. :-)

**Review by ktock — APPROVED** [review]

**Review by ktock — APPROVED** [review]

**Comment by yankay** [comment]
Thanks @fahedouch , @ktock
for the PR review

**Pull request merged**